tests/test_datasets/test_dataset.py

```diff
@@ -623,14 +623,11 @@ def test_to_flatten_with_exog(df_and_regressors_flat):
     for column, dtype in dtypes.items():
         if dtype == "category":
             expected_df[column] = expected_df[column].astype(dtype)
-
+    # reindex df to assert correct columns order
+    expected_df = expected_df.reindex(columns = ["timestamp", "segment", "target"] + list(expected_df.columns[:5]))
```

Comment:
Let's try to avoid this line and modify `sorted_columns` instead, I think it will be more clear.